epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyVisibilityItem.java

```diff
@@ -55,31 +58,22 @@
    *
    * @param view        the current {@link com.airbnb.epoxy.EpoxyViewHolder}'s itemView
    * @param parent      the {@link androidx.recyclerview.widget.RecyclerView}
-   * @param vertical    true if it scroll vertically
    * @return true if the view has been measured
    */
-  boolean update(@NonNull View view, @NonNull RecyclerView parent,
-      boolean vertical, boolean detachEvent) {
+  boolean update(@NonNull View view, @NonNull RecyclerView parent, boolean detachEvent) {
     // Clear the rect before calling getLocalVisibleRect
     localVisibleRect.setEmpty();
-    boolean visibleRect = view.getLocalVisibleRect(localVisibleRect);
-    this.verticalScrolling = vertical;
-    if (vertical) {
-      sizeInScrollingDirection = view.getMeasuredHeight();
-      sizeNotInScrollingDirection = view.getMeasuredWidth();
-      viewportSize = parent.getMeasuredHeight();
-      visibleSize = detachEvent || !visibleRect ? 0 : localVisibleRect.height();
-    } else {
-      sizeNotInScrollingDirection = view.getMeasuredHeight();
-      sizeInScrollingDirection = view.getMeasuredWidth();
-      viewportSize = parent.getMeasuredWidth();
-      visibleSize = detachEvent || !visibleRect ? 0 : localVisibleRect.width();
-    }
-    percentVisibleSize = detachEvent ? 0 : 100.f / sizeInScrollingDirection * visibleSize;
-    if (visibleSize != sizeInScrollingDirection) {
+    boolean viewDrawn = view.getLocalVisibleRect(localVisibleRect) && !detachEvent;
+    height = view.getHeight();
+    width = view.getWidth();
+    viewportHeight = parent.getHeight();
+    viewportWidth = parent.getWidth();
+    visibleHeight = viewDrawn ? localVisibleRect.height() : 0;
+    visibleWidth = viewDrawn ? localVisibleRect.width() : 0;
+    if (visibleHeight != height || visibleWidth != width) {
       fullyVisible = false;
```

Comment:
I know this is from a previous PR, but it's confusing to me that `fullyVisible` is set to false here, but it is only set to true in another method - would it be possible to do all assignment in one place?

Reply:
Yes, I will look into that. I am 100% is simplifying the code and this is a good opportunity now.